lib/interfaces/include/IOExpander.h

```diff
@@ -0,0 +1,25 @@
+#ifndef IOEXPANDER_H
+#define IOEXPANDER_H
+
+#include <etl/singleton.h>
+#include "MCP23017.h"
+
+class IOExpander
```

Comment:
Having separate read() and getBit() functions is fine, but since doing a full "update" of a value requires two function calls (read() then getBit()), I'd make special note of that in the class documentation header.